src/interfaces.py

```diff
@@ -1,5 +1,105 @@
 from collections.abc import Callable
-from typing import Protocol
+from enum import Enum
+from typing import List, Protocol, Tuple
+
+
+class ISimulationObject(Protocol):
+    """An object synchronized with the simulation"""
+
+    traci_id: str
+
+    def update(self, updates: dict) -> None:
+        """Gets called whenever a simualtion tick has happened.
+
+        :param updates: The updated values for the synchronized properties
+        """
+
+    def add_subscriptions(self) -> int:
+        """Gets called when this object is created to allow
+        specification of simulator-synchronized properties.
+
+        :return: The synchronized properties (see <https://sumo.dlr.de/pydoc/traci.constants.html>)
+        """
+
+
+class ISimulationNode(ISimulationObject, Protocol):
```

Comment:
Doesn't get `Protocol` inherited via `ISimulationObject` to `ISimulationNode`(and the other classes) too? 🤔

Reply:
I am not sure. I read [this post](https://stackoverflow.com/questions/69783371/python-multiple-inheritance-from-class-and-protocol), but re-reading it does not state that one would need to use multi-inheritance ...

Reply:
When you inherit from a Protocol the resulting subclass won't be a Protocol anymore. Therefore if you want to have it to be a Protocol you have to explicitly inherit from it again.

See https://peps.python.org/pep-0544/#merging-and-extending-protocols

Reply:
Ahhhh...gracias! 💚